core/state-machine/src/overlayed_changes.rs

```diff
@@ -115,14 +122,15 @@ impl OverlayedChanges {
 	/// Returns a double-Option: None if the key is unknown (i.e. and the query should be refered
```

Comment:
I know its from before this PR, but a custom enum like `OverlayValue::{NotFound, Deleted, Modified}` would make this code easier to follow, compared to the doubly nested `Option`.